Vreijs/functions3

vreijs opened the pull request:
don;t know what I now did;-)  I pressed 'New pull request' does t hat mean that you need to pull it? Or that I need to pull (if I do that then RStudio tell all is uptodate (which I would expect;-)> So to whom to I issue this New pull request and to what aim (that you then merge it? Sorry for asking these questions, it helps my understanding.

vreijs committed:
added swe_heliacal_angle

vreijs committed:
an error in de return description of swe_heliacal_angle

vreijs committed:
include swe_juldate function and tried to include swe_calc_ut, but I am not able to get it recognised (although I use same code, I think, as swe_calc

vreijs committed:
added swe_fixstar2_ut and updated correctly expert-interface.R

vreijs committed:
added swe_pheno

rstub commented on R/RcppExports.R:
```diff
@@ -84,6 +84,18 @@ swe_set_ephe_path <- function(path) {
 
+#' Determine Julian date number from calendar date
+#' @param year  Year
+#' @param mont  Month
```
Typo: `month` with an `h`

rstub commented on R/RcppExports.R:
```diff
@@ -251,3 +281,17 @@ swe_topo_arcus_visionis <- function(tjd_ut, dgeo, datm, dobs, helflag, mag, AziO
+#' @param AziO  The object's azimut
+#' @param AziS  The sun's azimut
+#' @param AziM  The moon's azimut
+#' @param AltM  The moon's altitude
```
You should not repeat these five argument descriptions here, since they have been mentioned with `swe_topo_arcus_visionis`. Since currently all documentation goes onto one page, one has to be careful to not repeat any arguments ...

vreijs committed:
rectified a type (mont shoudl be month

vreijs committed:
removed duplicated params

rstub commented:
> don;t know what I now did;-) I pressed 'New pull request' does t hat mean that you need to pull it? Or that I need to pull (if I do that then RStudio tell all is uptodate (which I would expect;-)> So to whom to I issue this New pull request and to what aim (that you then merge it? Sorry for asking these questions, it helps my understanding.

What you did was fine. Technically you are requesting me to `fetch` your changes and `merge` them into a `master` branch. Now a `fetch` followed by a `merge` is exactly what `pull` does, hence the name "pull request".

One can open such PRs even before the code is ready, since it gives a nice place to discuss the changes.

vreijs commented:
I am slowly understand the workflow. I sometimes though don;t understand why I need to roxygen2::roxygenise(): but I now do it mostly when I added a new function (and that seems to do a trick). But sometimes I don't understand why that also starts a compiler session: while just before this roxygenise I did a 'Install and Rebuild'. Is that because there is a kind of two-pass compiling needed...

rstub commented:
Currently the "Phenomenology info from Moon (UT)" Test fails on Travis/Linux, c.f. https://travis-ci.org/rstub/swephR/jobs/426914658#L1184 ff. It passes on Travis/MacOS and Appveyor/Windows. Main difference is that on Travis/Linux the sewphRdata pacakge is installed.

rstub committed:
Tell RStudio to use roxygen [skip ci]

rstub commented:
Concerning roxygen: RStudio can do that automatically, c.f. https://support.rstudio.com/hc/en-us/articles/200532317-Writing-Package-Documentation and my mini-commit. I would also check "Build & Reload".

vreijs commented:
I checked that option in the past, but RStudio keep hanging on me...
I need to check what the precise differences are between 'Load ALL',
Install and Restart' and 'Built & Reload'.

rstub commented:
In http://r-pkgs.had.co.nz/src.html#src-workflow the `devtools` author advocates using "Install and Restart" over "Load All" when using compiled code.

vreijs commented:
Thanks for this link

> In http://r-pkgs.had.co.nz/src.html#src-workflow the devtools author
> advocates using "Install and Restart" over "Load All" when using compiled
> code.
>

vreijs committed:
removed some typos

vreijs committed:
Merge branch 'vreijs/functions3' of https://github.com/rstub/swephR into vreijs/functions3

rstub committed:
Have swe_pheno() return 'iflag' instead of 'OK'

Most calculation functions in SE return the flag that is used in
the actual calculations. This was not the case for swe_pheno()
causing problems with swe_pheno_ut(). (fixes 20)

rstub commented:
Do you know of any functions missing for skyscapeR? I would like to merge this branch soon and start with the restructuring of source and documentation into several files. This schouln’t take long.

vreijs commented:
There are a few astrolibR functions that skyscapeR uses, which are not
(externally) documented within the SE. I made an overview:
https://github.com/f-silva-archaeo/skyscapeR/issues/4
Most of the function called in skyscapeR might not be needed to be
explicitly called as they are integrated in SE/swephR (like nutation,
aberration, ten, precess and cr2lst): I though need to check that by doing
the integration! The one function that is not in the SE documentation is
refraction, but I think we can make that function external (it is there as
I contributed to it;-). Remaining functions are certainly in
ARCHAEOCOSMO (hadec2altazi
and hor2eq), but if possible I want not to have ARCHAEOCOSMO yet linked in
skyscapeR (although in the near future it will IMHO due to heliacal events
and ray-tracing methods for refraction). I can also add these two function
in skyscapeR just as R script (they are very simple). When do you want to
merge it? I might have some time on coming Thursday or Friday to work on
this, but not today.

rstub commented:
Thanks for the update. If I find the time this evening, I will simply do the merge and restructuring. After that I can open a new branch for you.

vreijs commented:
Sounds ok.

rstub commented:
PR rebased and merged via CLI

rstub commented:
Basic restructuring is done. I have also created a new branch vreijs/functions4
